src/ServiceControl.Audit/Infrastructure/RavenDB/MigrateKnownEndpoints.cs

```diff
@@ -0,0 +1,88 @@
+﻿using NServiceBus;
+using NServiceBus.Features;
+using Raven.Client;
+using ServiceControl.Audit.Monitoring;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceControl.Audit.Infrastructure.RavenDB
+{
+    class MigrateKnownEndpoints : Feature
+    {
+        public MigrateKnownEndpoints()
+        {
+            EnableByDefault();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.RegisterStartupTask(b => new KnownEndpointMigrator(b.Build<IDocumentStore>()));
+        }
+    }
+
+    class KnownEndpointMigrator : FeatureStartupTask
+    {
+        IDocumentStore store;
+
+        public KnownEndpointMigrator(IDocumentStore store)
+        {
+            this.store = store;
+        }
+
+        protected override async Task OnStart(IMessageSession messageSession)
+        {
+            var knownEndpointsIndex = await store.AsyncDatabaseCommands.GetIndexAsync("EndpointsIndex").ConfigureAwait(false);
+            if (knownEndpointsIndex == null)
+            {
+                return;
+            }
+
+            await WaitForNonStaleIndex(knownEndpointsIndex.Name).ConfigureAwait(false);
```

Comment:
Isn't it fair to assume we can even operate on stale indexes? Given the index is a fanout index I'm sceptic it is good to wait for it to be non-stale. I would say the migration can be a best effort and we assume it to catch up over time

Reply:
The problem is the index will be deleted at the end of this code, so if it's stale it will never be caught up. We could wait for an hour before proceeding to give it a decent chance to update but not completely block.

Reply:
On a system with continuous load it will never be caught up. I think it is to dangerous to assume we ever can catch up

Reply:
It now waits an hour before giving up and just yolo'ing through the list

Reply:
That would hold up SC Audit from starting fro over an hour. That doesn't seem to be a good solution. Do it entirely in the background?

Reply:
I think it's fine to run off of non-stale data. The known endpoints collection likely doesn't change very often anyway. The intent of doing the migration is to try and capture the state before the switch. The stale index _is_ the state before the switch. The only case we aren't covering is the one where a endpoint has been added but has not yet appeared in the index. Chances are, that endpoint will still be sending/processing messages and will get picked up in the first few minutes of execution. In the rare case that an endpoint is not yet in the index and is no long sending/processing messages then I think it's OK for us to miss it.

Reply:
It's now an installer